libs/shared/typescript/charts/src/lib/utils/boxplot-utils.spec.ts

```diff
@@ -0,0 +1,364 @@
+          ['blue', 'red', 'yellow']
+        )
+      ).toEqual('red');
+      expect(consoleWarnSpy).not.toHaveBeenCalled();
+    });
+
+    it('should return default when point category is not found', () => {
+      expect(
+        getCategoryPointColor(
+          { xAxisCategory: 'b', pointCategory: 'WW', value: 1 },
+          ['X', 'Y', 'Z']
+        )
+      ).toEqual('transparent');
+      expect(consoleWarnSpy).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('getUniqueValues', () => {
+    it('should return sorted, unique values when specified', () => {
+      expect(
+        getUniqueValues(pointsWithPointCategories, 'xAxisCategory', true)
+      ).toEqual(['a', 'b', 'c']);
+
+      expect(
+        getUniqueValues(pointsWithPointCategories, 'pointCategory', true)
+      ).toEqual(['X', 'Y', 'Z']);
+    });
+
+    it('should return unsorted, unique values by default', () => {
+      expect(
+        getUniqueValues(pointsWithPointCategories, 'xAxisCategory')
+      ).toEqual(['c', 'a', 'b']);
+
+      expect(
+        getUniqueValues(pointsWithPointCategories, 'pointCategory')
+      ).toEqual(['Y', 'X', 'Z']);
+    });
+
+    it('should handle all undefined values', () => {
+      expect(
+        getUniqueValues(pointsWithoutPointCategories, 'pointCategory')
+      ).toEqual([]);
+    });
+  });
+
+  describe('getJitteredXValue', () => {
+    it('1 point', () => {
+      expect(getJitteredXValue(2, 0, 1)).toEqual(2);
+    });
+
+    it('2 points', () => {
+      expect(getJitteredXValue(2, 0, 2)).toEqual(1.9);
+      expect(getJitteredXValue(2, 1, 2)).toEqual(2.1);
+    });
+
+    it('3 points', () => {
+      expect(getJitteredXValue(2, 0, 3)).toEqual(1.8);
+      expect(getJitteredXValue(2, 1, 3)).toEqual(2);
+      expect(getJitteredXValue(2, 2, 3)).toEqual(2.2);
+    });
+
+    it('4 points', () => {
+      expect(getJitteredXValue(2, 0, 4)).toEqual(1.7);
+      expect(getJitteredXValue(2, 1, 4)).toEqual(1.9);
+      expect(getJitteredXValue(2, 2, 4)).toEqual(2.1);
+      expect(getJitteredXValue(2, 3, 4)).toEqual(2.3);
+    });
+
+    it('prevents overlap with next category when there are more subcategories than can be spaced using default spacing', () => {
+      expect(getJitteredXValue(2, 0, 6)).toEqual(1.55);
+      expect(getJitteredXValue(2, 5, 6)).toEqual(2.45);
+    });
+  });
+
+  describe('addXAxisValueToCategoryPoint', () => {
+    it('handles points without pointCategory values', () => {
+      const valuePoints: CategoryAsValuePoint[] = [
+        { xAxisCategory: 'c', value: 3, xAxisValue: 3 },
+        { xAxisCategory: 'a', value: 1, xAxisValue: 1 },
+        { xAxisCategory: 'b', value: 2, xAxisValue: 2 },
+      ];
+
+      const updatedPoints = addXAxisValueToCategoryPoint(
+        pointsWithoutPointCategories,
+        ['a', 'b', 'c'],
+        []
+      );
+      valuePoints.forEach((valuePt, index) => {
+        expect(valuePt.xAxisValue).toBeCloseTo(updatedPoints[index].xAxisValue);
+      });
```

Comment:
There is a slight rounding error, so use `toBeCloseTo` to compare the xAxisValues rather than directly comparing the objects with `toEqual`